# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Realm.h
# Commit: 33dcf5c2e643
# Full Hash: 33dcf5c2e643b94e90a0a197c99112a080501a12
# Author: Alexandra Borovova <aborovova@mozilla.com>
# Date: 2025-08-20 09:57:21
# Regressor Bug: 1980211
# File Overlap Count: 2
# Description:
#   Bug 1980211 - Attach locale override to Realm instead of setting it in the process. r=dom-core,smaug,tschuster
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D260169
# ==============================================================================

diff -r 2cb7ca176216 -r 33dcf5c2e643 js/src/vm/Realm.h
--- a/js/src/vm/Realm.h	Tue Aug 19 16:57:59 2025 +0000
+++ b/js/src/vm/Realm.h	Tue Aug 19 17:01:50 2025 +0000
@@ -508,6 +508,10 @@
     behaviors_.setReduceTimerPrecisionCallerType(type);
   }
 
+  void setLocaleOverride(const char* locale) {
+    behaviors_.setLocaleOverride(locale);
+  }
+
   /* Whether to preserve JIT code on non-shrinking GCs. */
   bool preserveJitCode() { return creationOptions_.preserveJitCode(); }
 